include/hipSYCL/sycl/libkernel/builtin_interface.hpp

```diff
@@ -53,6 +53,11 @@ HIPSYCL_BUILTIN T __hipsycl_acos(T x) noexcept {
   HIPSYCL_RETURN_DISPATCH_BUILTIN(__hipsycl_acos, x);
 }
 
+template<class T>
+HIPSYCL_BUILTIN int __hipsycl_isnan(T x) noexcept {
```

Comment:
Per the spec, the return type would vary: 32-bit int for float and 64-bit int for double. The easiest way to accomplish this is probably to have auto return type here, or, we could also introduce the convention that our builtins are always 32 bit, and then handle the correct type in the main builtins.hpp header.